CAP/gap/OppositeCategory.gi

```diff
@@ -38,6 +38,27 @@ InstallMethod( Opposite,
     
 end );
 
+##
+InstallOtherMethod( Opposite,
+                    [ IsCapCategory, IsCapCategoryMorphism ],
+                    
+  function( opposite_cat, morphism )
+    local source, range;
+    
+    if not HasOpposite( opposite_cat ) then
```

Comment:
I think this check is not what we want: If you compute `Opposite( C )`, the also `C` as the attribute `Opposite`. You could check for `WasCreatedAsOppositeCategory`, but I would suggest to simply remove this check and rely on `ObjectConstructor` and `MorphismConstructor` for checks.